yt/loaders.py

```diff
@@ -97,6 +97,21 @@ def load(
     if not fn.startswith("http"):
         fn = str(lookup_on_disk_data(fn))
 
+    # import it here cause it's costly
+    from importlib.metadata import entry_points
+
+    # overly complicated dance to suppress spurious warning
+    # https://github.com/python/importlib_metadata/pull/278
+    entrypoints = entry_points()
+    if hasattr(entrypoints, "select"):
```

Comment:
By making this conditional explicitly about the runtime Python version number we enable auto-cleaning with pyupgrade in the future
```suggestion
    if sys.version_info >= (3, 10):
```

Reply:
I discovered the minimal required version manually, but it's also documented in https://docs.python.org/3.11/library/importlib.metadata.html?highlight=importlib%20metadata#entry-points
(see Compatibility Note)

Reply:
the paragraph further reads 
> In the future, calling entry_points with no parameters will return an EntryPoints object. Users should rely on the selection interface to retrieve entry points by group.

I guess this means that to maximise stability, we should retrieve `external_frontends = entry_points(group="yt.frontends")` directly in Python 3.10 and newer.

Reply:
yeah, but now I have to figure out mocking for two versions of python along with a fix for windows...

Reply:
Good luck. I may be able to give you a hand a couple hours from now if you need.